Simplify Pull Request Template

mindyls opened the pull request:
# Related Issue
fixes NRLMMD-GEOIPS/geoips#17

# Testing Instructions
Once PR template is merged to main, open pull request in any NRLMMD-GEOIPS repo to view new automated template.
Updated template: https://github.com/NRLMMD-GEOIPS/.github/blob/mindyls-patch-1/.github/pull_request_template.md
review-template.md: https://github.com/NRLMMD-GEOIPS/.github/blob/mindyls-patch-1/.github/review-template.md

# Summary
## NRLMMD-GEOIPS/geoips#17: 2022-08-12, simplify PR template

### Documentation Updates
* **pull_request_template.md**
    * Move "Reviewer Checklist" to bottom of PR template
    * Remove <issue_id> and <reponame> from PR titles
        * Since PR contents are used directly in squashed commit messages, shorten PR title
    * Remove large instruction blocks
    * Remove example formats from Reviewer Checklist (title and issue)

# Output
This PR follows new template.

# Reviewer Checklist

### Ensure you are logged into GitHub

### Confirm all requirements are met for pull request
* Ensure appropriate / sufficient content
    * **Title format**: geoips#<issue_num> <modified_repo_name> - \<Short description>
    * **Included Issue**: Link to related Issue included at the beginning of pull request
    * **CHANGELOG updated**: Functionality included in pull request is ALSO referenced in CHANGELOG.md
    * **Demonstrated Testing**: Proper testing / output was demonstrated for functionality updates
    * **Included Outputs**: Imagery is included in "Output" section for new or changed product outputs (for reports!)
    * **Test Scripts/Outputs for New Functionality**: If new functionality is included in PR,
        ensure related test scripts and test outputs were included.
* Ensure all appropriate tags / attributes added along right-hand side of pull request
    * **Label**: Added appropriate descriptors
    * **Projects**: GeoIPS - All Repos and All Functionality
        * Other projects allowed as appropriate
* Ensure Related Issue is finalized appropriately (follow link above)
    
### Once all items in checklist have been confirmed:
* **Approve pull request**
    * Click "Files changed" tab
    * Click green "Review changes" button
    * Select "Approve" option
    * Add message if desired
    * Click green "Submit review" button
    * Project status will automatically be updated in Project "GeoIPS - All Repos and All Functionality" when pull request is approved.

jsolbrig commented:
My two cents:
- Can we update "Related Issue" to "Related Issues" so that multiple can be included when needed?
- Should the instructions be enclosed in block comments rather than using brackets? Something like this?
  ```
  <!---
  Please add one of the following here:
  - A link to ticket with testing instructions
  - A note that no exhaustive testing is required (if you have sufficient output below to demonstrate success, and it will be fully tested with next release)
  - Full instructions for testing this change
  --->
  ```
  This results in the instructions showing up in the template, but not in the final result if they aren't deleted.
- Maybe he checklist should be converted to be true checklists using `- [ ]` and we shouldn't allow this to pass until everything has been checked off?
  - I can look into an action to block merging until the check boxes have been clicked.

mindyls commented:
> My two cents:
> 
> * Can we update "Related Issue" to "Related Issues" so that multiple can be included when needed?
> * Should the instructions be enclosed in block comments rather than using brackets? Something like this?
>   ```
>   <!---
>   Please add one of the following here:
>   - A link to ticket with testing instructions
>   - A note that no exhaustive testing is required (if you have sufficient output below to demonstrate success, and it will be fully tested with next release)
>   - Full instructions for testing this change
>   --->
>   ```
>   
>   
>       
>         
>       
>   
>         
>       
>   
>       
>     
>   This results in the instructions showing up in the template, but not in the final result if they aren't deleted.
> * Maybe he checklist should be converted to be true checklists using `- [ ]` and we shouldn't allow this to pass until everything has been checked off?
>   
>   * I can look into an action to block merging until the check boxes have been clicked.

* Converted brackets to comments
* Moved Reviewer Checklist to separate review-template.md file per https://github.com/github/docs/edit/main/.github/review-template.md
* Update Related Issue to Related Issues

jsolbrig reviewed (approved):
I think this looks good. I made one small change to update "Related Issue" to be plural.
